flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/metadata/SelectivityEstimator.scala

```diff
@@ -541,7 +541,14 @@ class SelectivityEstimator(rel: RelNode, mq: FlinkRelMetadataQuery)
     }
     val columnInterval = mq.getColumnInterval(rel, inputRef.getIndex)
     if (columnInterval == null) {
-      return defaultEqualsSelectivity
+      // For types like VARCHAR, it has no 'interval' concept but may also contain different values.
+      // We utilize ndv to get a better estimate.
+      val ndv = mq. getDistinctRowCount(rel, ImmutableBitSet.of(inputRef.getIndex), null)
```

Comment:
nit: remove blank after mq.

Reply:
Can I confirm getDistinctRowCount is accessing existing statistics and is not doing anything non-performant?

Reply:
Hi thanks for the review! Yes, getDistinctRowCount is accessing existing statistics and is not doing anything non-performant. In fact, when `convertedInterval` is defined, you can see that `getDistinctRowCount` would be called as well to get `ndv`.